src/main/kotlin/org/rust/ide/inspections/fixes/RemoveTypeParameter.kt

```diff
@@ -0,0 +1,37 @@
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.openapi.project.Project
+import org.rust.lang.core.psi.RsBaseType
+import org.rust.lang.core.psi.RsCallExpr
+import org.rust.lang.core.psi.RsMethodCallExpr
+import org.rust.lang.core.psi.RsPathExpr
+import org.rust.lang.core.psi.ext.RsCompositeElement
+
+class RemoveTypeParameter(val element: RsCompositeElement) : LocalQuickFix {
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val typeArgumentList = when (element) {
+            is RsMethodCallExpr -> {
+                element.typeArgumentList
+            }
+            is RsCallExpr -> {
+                val pathExpr = element.expr as RsPathExpr?
+                pathExpr?.path?.typeArgumentList
+            }
+            is RsBaseType -> {
+                element.path?.typeArgumentList
+            }
+            else -> {
+                null
+            }
+        } ?: return
+        val lifetime = typeArgumentList.lifetimeList.size
+        if (lifetime == 0) {
+            typeArgumentList.delete()
+        }
+    }
+
+    override fun getName() = "Remove all type parameter"
+    override fun getFamilyName() = name
```

Comment:
We usually put properties at the beginning of the class, before methods, so it would be easier to find them.
Also a typo: parameter**s**